certbot-nginx/certbot_nginx/configurator.py

```diff
@@ -443,15 +470,29 @@ def _make_server_ssl(self, vhost):
         :type vhost: :class:`~certbot_nginx.obj.VirtualHost`
 
         """
+        ipv6info = self.ipv6_info(self.config.tls_sni_01_port)
+
         # If the vhost was implicitly listening on the default Nginx port,
         # have it continue to do so.
         if len(vhost.addrs) == 0:
             listen_block = [['\n    ', 'listen', ' ', self.DEFAULT_LISTEN_PORT]]
             self.parser.add_server_directives(vhost, listen_block, replace=False)
 
+        ipv6_block = ['']
+        if vhost.ipv6_enabled():
+            ipv6_block = ['\n    ',
+                            'listen',
+                            ' ',
+                            '[::]:{0} ssl'.format(self.config.tls_sni_01_port)]
+            if not ipv6info[1]:
+                # ipv6only=on is absent in global config
+                ipv6_block.append(' ')
+                ipv6_block.append('ipv6only=on')
```

Comment:
Is there any reason not to always append `ipv6only=on`? It will sometimes be unnecessary, but it seems more robust to always explicitly include it instead of relying on existing things in the server block.

Reply:
ipv6only=on can unfortunately exist only once per IP:PORT pair, duplicate entries result into nginx not being able to start.

Reply:
awesome, classic